docs/.vuepress/theme/components/__tests__/Header.spec.js

```diff
@@ -0,0 +1,48 @@
+import { shallowMount } from '@vue/test-utils'
+import Header from "@/theme/components/Header"
+
+describe('Header', () => {
+  test('render header with searchbox', () => {
+    const wrapper = shallowMount(Header, {
+      mocks: {
+        $site: {
+          themeConfig: {
+            search: true
+          }
+        }
+      },
+      stubs: ['router-link']
+    })
+    expect(wrapper.element).toMatchSnapshot()
+  })
+
+  test('render header without searchbox', () => {
+    const wrapper = shallowMount(Header, {
+      mocks: {
+        $site: {
+          themeConfig: {
+            search: false
+          }
+        }
+      },
+      stubs: ['router-link']
+    })
+    expect(wrapper.element).toMatchSnapshot()
+  })
+
+  // test('toggle mode', () => {
```

Comment:
We should remove this test case, because the **toggle** action is actually not testing this component, but emit `toggle-mode`, which means it's not a unit test case for this component. 

**Header.vue**: line 26

```vue
<span class="pointer view-mode" @click="$emit('toggle-mode')">
```